Bug with missing links when passing entire dict

As discussed with @superstar54 in person, when I run the following code

```
    scf_outputs = task(run_scf)(
        metadata={'call_link_label': 'pw-scf'},
        atoms=kwargs['atoms'],
        parameters=scf_parameters,
        kpoints=kwargs['kpoints']['explicit_grid']
    )
```
the code complains that 
```
E           ValueError: Missing required inputs:
E             • pw-scf.parameters.control
E             • pw-scf.parameters.electrons
E             • pw-scf.parameters.system
```

whereas if I do

```
   scf_outputs = task(run_scf)(
        metadata={'call_link_label': 'pw-scf'},
        atoms=kwargs['atoms'],
        parameters={'control': scf_parameters['control'], 'system': scf_parameters['system'], 'electrons': scf_parameters['electrons']},
        kpoints=kwargs['kpoints']['explicit_grid']
   )
```

I get no such error

--- a/docs/gallery/advanced/autogen/context_manager.py
+++ b/docs/gallery/advanced/autogen/context_manager.py
@@ -529,16 +529,15 @@ with WorkGraph('AddMap') as wg:
             y=get_value(map_zone.item.value, 'y').result,
         ).result
         map_zone.gather({'result': result})
-        wg.ctx.result = map_zone.outputs.result
-        wg.outputs.result = wg.ctx.result
+        wg.outputs.result = map_zone.outputs.result
 
 wg.run()
 
 print('\nResults:')
-for i, item in enumerate(wg.outputs.result.value.values()):
-    print(f'  Item {i}: {item}')
+for item in wg.outputs.result:
+    print(f'  {item._name}: {item.value}')
 # (1+1) + (2+2) + (3+3) = 12
-assert sum(wg.outputs.result.value.values()) == 12
+sum([socket.value.value for socket in wg.outputs.result]) == 12
 
 # %%
 # Let's inspect the task graph:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -19,7 +19,7 @@ classifiers = [
 dependencies = [
   "numpy",
   "scipy",
-  "node-graph~=0.6.2",
+  "node-graph~=0.6.3",
   "node-graph-widget>=0.0.5",
   "aiida-core~=2.7.1",
   "cloudpickle",

--- a/src/aiida_workgraph/tasks/builtins.py
+++ b/src/aiida_workgraph/tasks/builtins.py
@@ -1,7 +1,7 @@
 from __future__ import annotations
 from typing import Any, Dict
 from aiida_workgraph.task import Task
-from aiida_workgraph import task, namespace, meta
+from aiida_workgraph import task, namespace, meta, dynamic
 from node_graph.tasks.builtins import _GraphIOSharedMixin
 from node_graph.task import ChildTaskSet
 from node_graph.socket import BaseSocket
@@ -95,7 +95,7 @@ class Map(Zone):
         task_type='MAP',
         catalog='Control',
         inputs=namespace(
-            source=SocketSpec('workgraph.any', link_limit=100000),
+            source=dynamic(Any),
         ),
         outputs=namespace(),
         base_class_path='aiida_workgraph.tasks.builtins.Map',
@@ -125,7 +125,7 @@ class Map(Zone):
         gather_item = self.gather_item_task
         for name in sockets:
             gather_item.add_input_spec('workgraph.any', name=name)
-            self.add_output_spec('workgraph.any', name=name)
+            self.add_output_spec('workgraph.namespace', name=name)
         gather_item.set_inputs(sockets)
         return gather_item.outputs
 

--- a/tests/test_pythonjob.py
+++ b/tests/test_pythonjob.py
@@ -149,8 +149,10 @@ def test_PythonJob_namespace_output_input(fixture_localhost, python_executable_p
     """Test function with namespace output and input."""
 
     # output namespace
+    add_spec = namespace(sum=Any, total=Any)
+    add_multiply_spec = namespace(add=add_spec, multiply=Any)
     out = namespace(
-        add_multiply=namespace(add=namespace(sum=Any, total=Any), multiply=Any),
+        add_multiply=add_multiply_spec,
         minus=Any,
     )
 
@@ -163,13 +165,13 @@ def test_PythonJob_namespace_output_input(fixture_localhost, python_executable_p
 
     # input namespace
     @task.pythonjob
-    def myfunc2(x, y):
+    def myfunc2(x: add_multiply_spec, y):
         add = x['add']['sum']
         multiply = x['multiply']
         return y + add + multiply
 
     @task.pythonjob
-    def myfunc3(x, y):
+    def myfunc3(x: add_spec, y):
         return x['sum'] + y
 
     wg = WorkGraph('test_namespace_outputs')

